crates/ty_python_semantic/src/semantic_index/builder.rs

```diff
@@ -1875,11 +1879,50 @@ impl<'ast> Visitor<'ast> for SemanticIndexBuilder<'_, 'ast> {
                         });
                     }
                     let scope_id = self.current_scope();
+                    if let Some(nonlocals) = self.nonlocals_by_scope.get(&scope_id)
+                        && nonlocals.contains(&symbol_id)
+                    {
+                        // TODO: This should be some sort of error?
+                    }
                     self.globals_by_scope
                         .entry(scope_id)
                         .or_default()
                         .insert(symbol_id);
                 }
+                // TODO: why is this here?
+                walk_stmt(self, stmt);
```

Comment:
It seems like this does nothing? Curious what's up here.

Reply:
sort of a similar principle in spirit to [flake8-bugbear](https://github.com/PyCQA/flake8-bugbear)'s B906 rule (not yet reimplemented in Ruff, alas)